lib/fetch/util.js

```diff
@@ -6,6 +6,9 @@ const { isBlobLike, toUSVString, ReadableStreamFrom } = require('../core/util')
 const assert = require('assert')
 const { isUint8Array } = require('util/types')
 
+const textEncoder = new TextEncoder()
+const textDecoder = new TextDecoder('utf-8', { ignoreBOM: true })
```

Comment:
Unfortunately, this is both incorrect and insecure, you can't reuse encoder/decoders between inflight requests because the data will become corrupted.

Reply:
You are totally right. I don't know how I missed this. Reverting this commit for now. Thanks for the review @mcollina